dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2723,14 +2723,19 @@ void MergeTreeData::movePartitionToDisk(const ASTPtr & partition, const String &
     if (!disk)
         throw Exception("Disk " + name + " does not exists on policy " + storage_policy->getName(), ErrorCodes::UNKNOWN_DISK);
 
-    for (const auto & part : parts)
+    parts.erase(std::remove_if(parts.begin(), parts.end(), [&](auto part_ptr) {
```

Comment:
Need to fix the style of lambda and maybe it would be more obvious to write the for loop with named flags like found_part_on_another_disk = false?